### ISSUE TITLE
Poor skullstripping/registration BOLD->T1w results

### ISSUE THREAD

**Issue by jdkent** [poldracklab/fmriprep#701]
There was extra dura mater included with the brain in the parietal area, as well as registration issues as already noted in #694, sharing this subject as per @chrisfilo's request from [this conversation](https://neurostars.org/t/fmriprep-spatial-normalization/762)

[download zip](https://osf.io/4tqa7/download)

fmriprep was run using our cluster via the following command:
```
singularity run -H ${HOME}/singularity_home -B /Shared/vosslabhpc:/mnt \
/Shared/vosslabhpc/UniversalSoftware/SingularityContainers/poldracklab_fmriprep_latest-2017-08-12-1b62280f96f1.img \
/mnt/Projects/PACR-AD/Imaging/BIDS /mnt/Projects/PACR-AD/Imaging/BIDS/derivatives \
participant --participant_label controlGE140 \
-w /mnt/Projects/PACR-AD/Imaging/BIDS/derivatives/work/fmriprep/ \
--write-graph --mem_mb 35000 --omp-nthreads 10 --nthreads 16 --use-aroma --bold2t1w-dof 6 \
--output-space fsnative template \
--template MNI152NLin2009cAsym
```

Let me know if you need any additional information.

**Comment by effigies** [comment]
Using the NKI template (see #729) had no discernible effect on the skull-stripping. And I tried manually running `mri_watershed` on the `T1.mgz` outputs in FreeSurfer, and found that it was extremely overzealous, removing most of prefrontal cortex.

I'm wondering if what we should do is permit users to edit the outputs and make a second pass, with `--fixed-skull-strip img.nii.gz`. I would basically make a check that the passed image has the same dimensions/affine, and re-run any downstream preprocessing, including patching the fixed skull-strip into FreeSurfer.

I would not by default re-run FreeSurfer.

**Comment by effigies** [comment]
Hi @jdkent. I've had a few pokes at this, but I don't think that we're going to be able to move very quickly here.

I think the fastest way to resolve this would be to move to the [ANTs discussion list](https://sourceforge.net/p/advants/discussion/840261/), and provide the commandline in your `anat_preproc_wf/skullstrip_ants_wf/t1_skull_strip/command.txt` file. Note that the `T_template0_*.nii.gz` files are from an [RAS-oriented version of their OASIS template](https://osf.io/bxuga/).

In any event, if you manage to fix this before we can, please let us know how, and we'll try to incorporate these fixes into fmriprep.

**Comment by oesteban** [comment]
Hi @jdkent how available would you be to help me implement `antsBrainExtraction.sh` in nipype, so that we have the flexibility to adapt processing to edge cases like yours?

I've been considering such nipype implementation for MRIQC since long, and maybe we could join forces here.

**Comment by oesteban** [comment]
Highly related to #730 as well

**Comment by jdkent** [comment]
Oops, didn't see this. @oesteban I can carve out some time for this, but isn't `antsBrainExtraction.sh` already [implemented in nipype](http://nipype.readthedocs.io/en/latest/interfaces/generated/interfaces.ants/segmentation.html#antsbrainextraction)?

I still need to get the edge case on the ANTs discussion list, so I can start there, and then see what changes to antsBrainExtraction.sh (if any) appear.

**Comment by oesteban** [comment]
Sorry, @jdkent I was too vague on my comment.

Yes, @rwblair wrote an interface for the original [`antsBrainExtraction.sh`](https://github.com/ANTsX/ANTs/blob/074d43632982b4473fcbd06233b43ec6a8e4f60b/Scripts/antsBrainExtraction.sh) script. That is a bash-script with several ants commands together. We would benefit a lot from reproducing it in nipype, particularly because we would have all intermediate results available without the need of modifying the nipype interface at all.

This is a similar need to porting FreeSurfer's `recon-all` into nipype: a long-lasting item in the wish list.

**Comment by effigies** [comment]
Hi @jdkent. Just checking in on this issue. Are you still experiencing this with recent fmriprep? I'm wondering whether the recent inclusion of FreeSurfer segmentations to refine the brainmask may have ended up resolving your issue.

Outside of that, have you had any luck skull-stripping these brains with other tools, or using some other configuration?

**Comment by jdkent** [comment]
I haven't tried again recently (i.e. still no progress), but I should have more time to devote to this in early February. Thank you for your patience.

**Comment by jdkent** [comment]
In my humble opinion, the registration do indeed look better, great work! I'll support my statement with some pictures tomorrow.

**Comment by oesteban** [comment]
Seems to be solved, feel free to reopen if not 👍

**Comment by oliver-xie** [comment]
Hello fmriprep experts, we recently encountered a similar issue with our T1 skull stripping. Part of dura mater was included in the brain mask leading to very poor EPI-T1 registration. 
![image](https://user-images.githubusercontent.com/3640318/53057761-83dab300-3465-11e9-9491-f0e09c71773f.png)
Using NKI template made the skull stripping even worse. I wonder if there is any solution to this. Thank you!

**Comment by effigies** [comment]
Hi, can you open a new issue? This one will be related to the [sMRIPrep](https://github.com/poldracklab/smriprep/issues/) (structural) workflows.
